test/jdk/valhalla/valuetypes/ObjectMethods.java

```diff
@@ -24,85 +24,128 @@
-    static final Line LINE2 = Line.makeLine(10, 20, 3, 4);
-    static final MutablePath MUTABLE_PATH = MutablePath.makePath(10, 20, 30, 40);
-    static final MixedValues MIXED_VALUES = new MixedValues(P1, LINE1, MUTABLE_PATH, "value");
-    static final InlinableValue VALUE = new InlinableValue.Builder()
-                                                .setChar('z')
-                                                .setBoolean(false)
-                                                .setByte((byte)0x1)
-                                                .setShort((short)3)
-                                                .setLong(4L)
-                                                .setPoint(Point.makePoint(200, 200))
-                                                .setNumber(InlinableValue.Number.intValue(10)).build();
-    static final InlinableValue VALUE1 = new InlinableValue.Builder()
-                                                .setChar('z')
-                                                .setBoolean(false)
-                                                .setByte((byte)0x1)
-                                                .setShort((short)3)
-                                                .setLong(4L)
-                                                .setPoint(Point.makePoint(100, 100))
-                                                .setPointRef(Point.makePoint(200, 200))
-                                                .setReference(Point.makePoint(300, 300))
-                                                .setNumber(InlinableValue.Number.intValue(20)).build();
-
-    @DataProvider(name="Identities")
-    Object[][] identitiesData() {
-        return new Object[][]{
-                {new Object(), false, false},
-                {"String", true, false},
-                {String.class, true, false},
-                {Object.class, true, false},
-                {new ValueType1(1), false, true},
-                {new ValueType2(2), false, true},
-                {new PrimitiveRecord(1, "A"), false, true},
-                {new ValueRecord(1,"B"), false, true},
-                {new int[0], true, false},  // arrays of primitive classes are identity objects
-                {new Object[0], true, false},  // arrays of identity classes are identity objects
-                {new String[0], true, false},  // arrays of identity classes are identity objects
-                {new ValueType1[0], true, false},  // arrays of value classes are identity objects
-        };
+    static final Point P1 = new Point(1, 2);
+    static final Point P2 = new Point(30, 40);
+    static final Line L1 = new Line(1, 2, 3, 4);
+    static final Line L2 = new Line(10, 20, 3, 4);
+    static final Ref R1 = new Ref(P1, L1);
+    static final Ref R2 = new Ref(P2, null);
+    static final Value V = new Value(P1, L1, R1, "value");
+
+    static Stream<Arguments> identitiesData() {
+        return Stream.of(
+                Arguments.of(new Object(), false, false),
```

Comment:
The JEP doesn't say this (yet) but `new Object()` is an identity instance. java.util.Objects.isIdentityObject() and .isValueObject() will need to be updated too.

Reply:
It already handles that.  The two arguments specify if the class of the instance is identity class or value class.     The test special cases to verify if the class is `Object.class`, it's an identity object.